router/http.go

```diff
@@ -166,15 +166,22 @@ func (s *HTTPListener) doSync(ctx context.Context, errc chan<- error) <-chan str
 
 func (s *HTTPListener) startListen() error {
 	if err := s.listenAndServe(); err != nil {
+		s.Close()
 		return err
 	}
-	s.Addr = s.listener.Addr().String()
+	s.Addrs = []string{}
```

Comment:
In general if you know how big you need a slice to be, you should make it that big to avoid Go needing to guess how big it needs to be the first time you `append`.